### PR TITLE
init: prebuild some binaries

### PR DESCRIPTION
**Pull request by rminnich**
There was concern about installcommand hanging around after
building the first rush. We experimented with using syscall.Exec
but that led to some other problems in the Go runtime.

Until such time as that is fixed, we can prebuild some of our
programs. When we do this, we get the clean ps shown below.

% ps ax
   PID TTY                                                TIME CMD
 26150 pts/0                                          00:00:00 sudo
 26151 pts/0                                          00:00:00 init
 27160 pts/0                                          00:00:00 rush

Note that our ps command has an unusually bad case of shitespace, which we'll need
to fix in a forthcoming PR. Given all those blanks, we might as well show PGID and TTY.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
So, I think Ryan is right again. We should move this project to Canada. I could swear this did not work last night, but it works now. ^C is working in QEMU. Please test.

File: cmds/init/init.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ var (
 	ludicrous = flag.Bool("ludicrous", false, "print out information about symlink creation")
 	test      = flag.Bool("test", false, "Test mode: don't try to set control tty")
 	debug     = func(string, ...interface{}) {}
+
+	// There are some targes it makes sense to prebuild, since at least one never
```
Comment (hugelgupf) [inline cmds/init/init.go]: s/targes/targets

File: cmds/init/init.go [hunk 2]
```diff
@@ -127,6 +132,14 @@ func main() {
 		}
 	}
 
+	// We can try to prebuild but it's not necessarily and error, since
+	// whoever put this initramfs together might have excluded things.
+	for _, v := range prebuild {
+		if o, err := exec.Command("installcommand", []string{"--onlybuild", v}...).CombinedOutput(); err != nil {
```
Comment (hugelgupf) [inline cmds/init/init.go]: So when does this usually happen? And why does it need to be prebuilt?

File: cmds/init/init.go [hunk 3]
```diff
@@ -138,7 +151,8 @@ func main() {
 	// Perhaps we should stat inito first.
 	// inito is always first and we set default flags for it.
 	cloneFlags := uintptr(syscall.CLONE_NEWPID)
-	for _, v := range []string{"/inito", "/buildbin/uinit", "/buildbin/rush"} {
+	for _, v := range append([]string{"/inito"}, prebuild...) {
+
```
Comment (hugelgupf) [inline cmds/init/init.go]: no newline here

File: cmds/init/init.go [hunk 4]
```diff
@@ -138,7 +151,8 @@ func main() {
 	// Perhaps we should stat inito first.
 	// inito is always first and we set default flags for it.
 	cloneFlags := uintptr(syscall.CLONE_NEWPID)
-	for _, v := range []string{"/inito", "/buildbin/uinit", "/buildbin/rush"} {
+	for _, v := range append([]string{"/inito"}, prebuild...) {
```
Comment (hugelgupf) [inline cmds/init/init.go]: more descriptive name than v?

File: cmds/init/init.go (hunk 2 above)
Comment (rminnich) [inline cmds/init/init.go]: I'm not even sure this is needed any more, now that we have syscall.Exec in installcommand. I"m gonna drop it.

**Comment by rminnich** [comment]
Exec works, so I don't see this as needed any more.

**Pull request closed without merging**